docker/scripts/cuda/runtime/install-vllm.sh

```diff
@@ -93,5 +94,10 @@ echo "DEBUG: Installing packages: ${INSTALL_PACKAGES[*]}"
 # install all packages in one command with verbose output to prevent GHA timeouts
 uv pip install -v "${INSTALL_PACKAGES[@]}"
 
+# uninstall the NVSHMEM dependency brought in by vllm if using a compiled NVSHMEM
+if [[ "${NVSHMEM_DIR-}" != "" ]]; then
+  uv pip uninstall nvidia-nvshmem-cu${CUDA_MAJOR} nvshmem4py-cu${CUDA_MAJOR}
```

Comment:
It looks like the right nvshmem4py-cu is installed (the one we built) so I will probably remove this before merge